File: 4th_year/1st_semester/cis492_graduation_project_II/just_bus/source_code/bus_api/Bus/Models/Driver.cs
using System;
using System.ComponentModel.DataAnnotations;
using System.ComponentModel.DataAnnotations.Schema;

#nullable disable

namespace Bus.Models
{
    public partial class Driver
    {
        [Key, Column(Order = 0)]
        public int PersonId { get; set; }
        [Key, Column(Order = 1)]
        public int VehicleId { get; set; }
        [Key, Column(Order = 2)]
        public int? RouteId { get; set; }
        public int? ImageId { get; set; }
        public decimal? LastLatitude { get; set; }
        public decimal? LastLongitude { get; set; }
        public DateTime? LastLocationUpdate { get; set; }
        public DateTime? ModificationDate { get; set; }
        public int? Modifier { get; set; }
        public DateTime? CreationDate { get; set; }
        public int? Creator { get; set; }

        public virtual Person CreatorNavigation { get; set; }
        public virtual Image Image { get; set; }
        public virtual Person ModifierNavigation { get; set; }
        public virtual Person Person { get; set; }
        public virtual Vehicle Vehicle { get; set; }

        public Driver ShowOnlyPersonName()
        {
            if (Person != null)
            {
                Person.Id = -1;
                Person.Number = null;
                Person.Password = null;
                Person.Salt = null;
                Person.Role = null;
            }

            return this;
        }
    }
}

File: 4th_year/1st_semester/cis492_graduation_project_II/just_bus/source_code/bus_flutter/lib/Assets/Models/driver.dart
import 'package:bus/Assets/Models/image_asset.dart';
import 'package:bus/Assets/Models/person.dart';
import 'package:bus/Assets/Models/vehicle.dart';

class Driver {
  late ImageAsset? image;
  late double lastLatitude;
  late double lastLongitude;
  late DateTime lastLocationUpdate;
  late Person person;
  late Vehicle vehicle;
  late int markerId;

  Driver(
      {required this.image,
      required this.lastLatitude,
      required this.lastLongitude,
      required this.lastLocationUpdate,
      required this.person,
      required this.vehicle});

  factory Driver.fromJson(Map<String, dynamic> parsedJson) {
    return Driver(
      image: parsedJson['image'] == null
          ? null
          : ImageAsset.fromJson(parsedJson['image']),
      lastLatitude: parsedJson['lastLatitude'],
      lastLongitude: parsedJson['lastLongitude'],
      lastLocationUpdate:
          DateTime.parse(parsedJson['lastLocationUpdate'].toString()),
      person: Person.fromJson(parsedJson['person']),
      vehicle: Vehicle.fromJson(parsedJson['vehicle']),
    );
  }

  Map<String, dynamic> toJson() => {
        'image': image == null ? null : image!.toJson(),
        'lastLatitude': lastLatitude,
        'lastLongitude': lastLongitude,
        'lastLocationUpdate': lastLocationUpdate,
        'person': person.toJson(),
        'vehicle': vehicle.toJson(),
      };
}
